### DISCUSSION TITLE
SERP preview rendering with a pink background?

### DISCUSSION THREAD

**Discussion by christaggart** [google/model-viewer#4119] [Q&A]
This question is a bit model-viewer-adjacent, but we have a gltf model that renders fine in the viewer itself, but when rendered as part of a SERP preview (3DModel schema), instead of the image texture being applied, it's rendering as a solid pink face? Does anyone know what might cause this?

<img width="256" alt="image" src="https://user-images.githubusercontent.com/205751/218503929-900caebc-4189-487e-96fa-7fe10dfb7ea7.png">

<img width="180" alt="image" src="https://user-images.githubusercontent.com/205751/218504864-9c9ab6f1-03e4-4138-b30a-e7ddc9697699.png">

**Comment by elalish** [comment]
@AdrianAtGoogle

**Reply by elalish** [reply]
@christaggart Can you point us to your URL or a search that brings this up, so we can repro?

**Reply by christaggart** [reply]
https://www.google.com/search?q=canvas+prints

**Reply by christaggart** [reply]
If it helps as well we're using 
```
$schema->ThreeDimensionalModel()->name('Canvas print')
                    ->encoding([
                        Schema::MediaObject()->contentUrl('https://d3n4dop8x7yrwi.cloudfront.net/glb/2023/01/10/5e0f96804a85672eb0da1b3981b8ca99/b8b150c094055dbafc5fb16fb04f4ae2.glb')->encodingFormat('model/gltf-binary'),
                        Schema::MediaObject()->contentUrl('https://d3n4dop8x7yrwi.cloudfront.net/usd/2023/01/10/b8b150c094055dbafc5fb16fb04f4ae2.usdz')->encodingFormat('model/vnd.usdz+zip')
                    ]);
```

**Reply by christaggart** [reply]
@elalish I did a bit more digging and I remembered that we switched from jpegs to webp for these models to massively improve filesize/download time for users. My guess would be whatever's getting used to render SERP previews doesn't yet support EXT_texture_webp?

https://github.com/KhronosGroup/glTF/blob/main/extensions/2.0/Vendor/EXT_texture_webp/README.md

<img width="1445" alt="image" src="https://user-images.githubusercontent.com/205751/220941425-24d74546-3386-409d-ab71-0d626fc0e12a.png">

**Reply by elalish** [reply]
Ah, that's a good bet! I believe Blender is being used for that generation internally, and not the most recent version either.

**Reply by victorjkhalaf** [reply]
Hey there!

We just updated the file, same filename (https://d3n4dop8x7yrwi.cloudfront.net/glb/2023/01/10/5e0f96804a85672eb0da1b3981b8ca99/b8b150c094055dbafc5fb16fb04f4ae2.glb) and confirmed it validates with image/jpeg. However we still see the pink model, presumably requiring a regeneration from your end. Is this something we can request a manual generation for somewhere?

**Reply by christaggart** [reply]
Can confirm it renders fine in Blender when using a jpeg. @AdrianAtGoogle
<img width="731" alt="image" src="https://user-images.githubusercontent.com/205751/224119464-215d8f75-26a1-4651-9a5f-7e06ed135e8e.png">
